File: src/main/java/org/umundo/model/Answer.java
package org.umundo.model;

import org.umundo.core.Message;

public class Answer {

  private String username;
  private int answer;
  private int questionId;

  public Answer(String username, int answer, int questionId) {
    this.username = username;
    this.answer = answer;
    this.questionId = questionId;
  }

  public static Answer fromMessage(Message m) {
    return new Answer(
        m.getMeta("username"),
        Integer.parseInt(m.getMeta("answer")),
        Integer.parseInt(m.getMeta("questionId"))
    );
  }

  public Message get() {
    Message m = new Message();
    m.putMeta("type", "answer");
    m.putMeta("username", username);
    m.putMeta("answer", answer + "");
    m.putMeta("questionId", questionId + "");
    return m;
  }

  public String getUsername() {
    return username;
  }

  public int getAnswer() {
    return answer;
  }

  public int getQuestionId() {
    return questionId;
  }
}

File: pyQuiz/answer.py
import config

class Answer:
    """Represents the answer to a quiz question."""

    def __init__(self, ui, question, answer_id):
        self._ui = ui
        self._q = question
        self._aid = answer_id

    def toDict(self):
        """
        Compatible representation of this object as dictonary.
        This representation should be used whenever the object is send over the
        wire via a umundo message.
        """
        return {
            "type": config.Message.ANSWER,
            "username": self._ui.username,
            "questionId": self._q.getQuestionId(),
            "answer": self._aid,
        }

